kernel/sched.c

```diff
@@ -222,10 +222,8 @@ static ALWAYS_INLINE void dequeue_thread(void *pq,
  */
 void z_requeue_current(struct k_thread *curr)
 {
-	LOCKED(&sched_spinlock) {
-		if (z_is_thread_queued(curr)) {
-			_priq_run_add(&_kernel.ready_q.runq, curr);
-		}
+	if (z_is_thread_queued(curr)) {
+		_priq_run_add(&_kernel.ready_q.runq, curr);
```

Comment:
Is this change because this spinlock here is now unnecessary, or because we now expect `z_requeue_current` to be called with `sched_spinlock` held?

One thing I will say as an "outsider" here is that it's decidedly non-obvious what functions are intended to be called with/without certain spinlocks held. (Ties into #32148 I suppose.)

(One possible future improvement here would be to add defines for REQUIRES_SPINLOCK_HELD / NOT_HELD that normally are nop-d out but expands to appropriate asserts based on a configuration flag.)

Reply:
The latter; lock management is now outside the function in the swap implementation.  And yes, it's a quick change so I didn't bother to do any refactoring to clarify the usage.